diff --git a/manifests/server.pp b/manifests/server.pp
index 9352f8d..e2ace63 100644
--- a/manifests/server.pp
+++ b/manifests/server.pp
@@ -336,7 +336,6 @@ class nagios::server (
   # Realize all nagios related exported resources for this server
   # Automatically reload nagios for relevant configuration changes
   # Require the package for the parent directory to exist initially
-  Nagios_command <<| tag == "nagios-${nagios_server}" |>>
   Nagios_contact <<| tag == "nagios-${nagios_server}" |>> {
     notify  => Service['nagios'],
     require => Package['nagios'],
@@ -345,7 +344,6 @@ class nagios::server (
     notify  => Service['nagios'],
     require => Package['nagios'],
   }
-  Nagios_host <<| tag == "nagios-${nagios_server}" |>>
   Nagios_hostdependency <<| tag == "nagios-${nagios_server}" |>> {
     notify  => Service['nagios'],
     require => Package['nagios'],
@@ -354,8 +352,6 @@ class nagios::server (
     notify  => Service['nagios'],
     require => Package['nagios'],
   }
-  Nagios_service <<| tag == "nagios-${nagios_server}" |>>
-  Nagios_servicedependency <<| tag == "nagios-${nagios_server}" |>>
   Nagios_servicegroup <<| tag == "nagios-${nagios_server}" |>> {
     notify  => Service['nagios'],
     require => Package['nagios'],
@@ -373,6 +369,11 @@ class nagios::server (
     require => Package['nagios'],
   }
 
+  Nagios_host <<| tag == "nagios-${nagios_server}" |>>
+  -> Nagios_command <<| tag == "nagios-${nagios_server}" |>>
+  -> Nagios_service <<| tag == "nagios-${nagios_server}" |>>
+  -> Nagios_servicedependency <<| tag == "nagios-${nagios_server}" |>>
+
   # Auto reload and parent dir, but for non-exported resources
   # FIXME: This does not work from outside here, wrong scope.
   # We'll need to wrap around these types with our own
